lib/kubeclient/common.rb

```diff
@@ -98,12 +104,7 @@ def discovery_needed?(method_sym)
     def handle_exception
       yield
     rescue RestClient::Exception => e
-      json_error_msg = begin
-        JSON.parse(e.response || '') || {}
-      rescue JSON::ParserError
-        {}
-      end
-      err_message = json_error_msg['message'] || e.message
+      err_message = json_error_message(e.response) || e.message
```

Comment:
@simon3z Because I used the same code for parsing the error message from the body of a `Signet::AuthorizationError`, so I encapsulated the code in a method.